### PR TITLE
add close connection when passing negative number to connection_timeout

### PR DESCRIPTION
**Pull request by magianxd**
Refer to issue #347.

### CODE REVIEW

File: pyVmomi/SoapAdapter.py [hunk 1]
```diff
@@ -1384,6 +1384,11 @@ def _CloseIdleConnections(self):
 
          for conn, _ in idleConnections:
             self._CloseConnection(conn)
+      else:
```
Comment (tianhao64) [inline pyVmomi/SoapAdapter.py]: This means we always close connection when the timeout is negative? But negative supposed to mean we never close connection. I agree that never close connection is a problem when the session times out for those existing connections. Maybe we should never allow a negative value?

Comment (magianxd) [inline pyVmomi/SoapAdapter.py]: Sounds good to me. I will put he verification code in SmartConnect since it is the entry point. Is that okay?

**Comment by jeking3** [comment]
The original author's fork/branch has been removed.  This can probably be closed.

**Comment by jeking3** [comment]
@tianhao64 The original author's fork/branch has been removed. This can probably be closed.

**Comment by jeking3** [comment]
@tianhao64 The original author's fork/branch has been removed. This can probably be closed.

**Comment by DanielDraganov** [comment]
Closing the request as agreed in https://github.com/vmware/pyvmomi/pull/518/commits/881e47023479874e80ecd241a29860d3a4024106#r104464420

**Pull request closed without merging**